### PR TITLE
Check Point: Added HA state support

### PR DESCRIPTION
**Pull request by lazyb0nes**
Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Review by calinrigo — APPROVED** [review]
It would be great to integrate the HA state support

**Comment by f0o** [comment]
@calinrigo have you been able to test these changes and verify that they work?

**Comment by lazyb0nes** [comment]
I am guessing someone else also has to check but i have at least two generations of Checkpoint that i've tested it on and it works as intended.

**Comment by calinrigo** [comment]
Just discovered that if you are running Checkpoint on Dell Appliances this HA State will not work.
Here is part of the discovery capture.

`NMP['/usr/bin/snmpget' '-v3' '-l' 'authNoPriv' '-n' "" '-a' 'MD5' '-A' 'PASSWORD' '-u' 'USER' '-OUQ' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/checkpoint' 'udp:HOSTNAME:161' 'CHECKPOINT-MIB::svnApplianceProductName.0' 'CHECKPOINT-MIB::svnApplianceSerialNumber.0' 'CHECKPOINT-MIB::svnVersion.0']

CHECKPOINT-MIB::svnApplianceProductName.0 = PowerEdge R740
CHECKPOINT-MIB::svnApplianceSerialNumber.0 = 
CHECKPOINT-MIB::svnVersion.0 = R80.30  
  

Yaml OS data: {"CHECKPOINT-MIB::svnApplianceProductName.0":"PowerEdge R740","CHECKPOINT-MIB::svnApplianceSerialNumber.0":"","CHECKPOINT-MIB::svnVersion.0":"R80.30"} 

Location: Unknown

Hardware: PowerEdge R740

OS Version: R80.30

OS Features: 

Serial: 


>> Runtime for discovery module 'os': 0.0630 seconds with 9304 bytes`

**Comment by calinrigo** [comment]
I think I got that wrong...
Checkpoint is a quite complicated beast.
I was testing your code againsts a checkpoint VSX cluster...

Trying to extend on what you did there on adding VSX cluster support.
As this is the first time doing this...I might not know what I'm doing here :P

**Comment by lazyb0nes** [comment]
This code will only work with Check Point VSX

VSX is only the hardware in this case i have tested it against:
**Check Point 23900 
Check Point 15600**

https://supportcenter.checkpoint.com/supportcenter/portal?eventSubmit_doGoviewsolutiondetails=&solutionid=sk90860&partition=General&product=Security

If you go under 
**VSX** and **Cluster**

**Comment by calinrigo** [comment]
There was a problem on my deployment.
It works now!
I have the HA state.
@f0o I can confirm it works on my side also.

**Comment by lazyb0nes** [comment]
> There was a problem on my deployment.
> It works now!
> I have the HA state.
> @f0o I can confirm it works on my side also.

Sot the HA state also works on a virtual Check Point appliance?

**Review by f0o — DISMISSED** [review]
LGTM

@librenms/testers if anyone has related devices and could double check that would be great. But it seems to be fine from above comments.

**Comment by epacke** [comment]
Need to update the old tests too I reckon. Coffee break is over. Will do that tonight.

**Comment by epacke** [comment]
@Jellyfrog About to break my keyboard in two and jump out of the window. It's on the bottom floor but it's COLD in Stockholm so it will still suck! GAH!!

I _think_ this is due to some locale or something like that. Seems like the rounding of numbers differ between my local instance and the one running the unit test. Have you seen this before?

**Comment by epacke** [comment]
Now it builds but the new sensors is not included in the json files. ** screaming into  a pillow **

**Comment by epacke** [comment]
Order is restored. I'm out of options now.

File: includes/definitions/discovery/gaia.yaml [hunk 1]
```diff
@@ -44,6 +44,39 @@ modules:
...
+                    oid: haState
+                    num_oid: '.1.3.6.1.4.1.2620.1.5.6.0.{{ $index }}'
+                    index: 'haState.{{ $index }}'
+                    descr: "Cluster member state"
+                    group: HA
+                    value: haState
+                    states:
+                        - { value: 1, descr: active , graph: 3, generic: 0 }
+                        - { value: 2, descr: standby, graph: 2, generic: 0 }
+                        - { value: 3, descr: ´active attention´, graph: 1, generic: 1 }
```
Comment (murrant) [inline includes/definitions/discovery/gaia.yaml:67]: the backticks here are not correct

File: includes/definitions/discovery/gaia.yaml [hunk 2]
```diff
@@ -44,6 +44,39 @@ modules:
                         - { value: 0, descr: 'false', graph: 1, generic: 0 }
                         - { value: 1, descr: 'true', graph: 1, generic: 2 }
                         - { value: 2, descr: 'reading error', graph: 1, generic: 3 }
+                -
+                    oid: haStarted
+                    num_oid: '.1.3.6.1.4.1.2620.1.5.5.0.{{ $index }}'
+                    index: 'haStarted.{{ $index }}'
+                    descr: "Cluster mechanism started"
```
Comment (murrant) [inline includes/definitions/discovery/gaia.yaml:51]: please use single quotes for consistency

File: includes/definitions/discovery/gaia.yaml [hunk 3]
```diff
@@ -44,6 +44,39 @@ modules:
...
+                    group: HA
+                    value: haStarted
+                    states:
+                        - { value: 1, descr: yes, graph: 1, generic: 0 }
+                        - { value: 0, descr: no, graph: 0, generic: 2 }
+                -
+                    oid: haState
+                    num_oid: '.1.3.6.1.4.1.2620.1.5.6.0.{{ $index }}'
+                    index: 'haState.{{ $index }}'
+                    descr: "Cluster member state"
```
Comment (murrant) [inline includes/definitions/discovery/gaia.yaml:61]: single quotes here

File: includes/definitions/discovery/gaia.yaml [hunk 4]
```diff
@@ -44,6 +44,39 @@ modules:
...
+                    states:
+                        - { value: 1, descr: active , graph: 3, generic: 0 }
+                        - { value: 2, descr: standby, graph: 2, generic: 0 }
+                        - { value: 3, descr: ´active attention´, graph: 1, generic: 1 }
+                        - { value: 0, descr: down, graph: 0, generic: 2 }
+                -
+                    oid: haStatCode
+                    num_oid: '.1.3.6.1.4.1.2620.1.5.101.{{ $index }}'
+                    index: 'haStatCode.{{ $index }}'
+                    descr: "Cluster member status code"
```
Comment (murrant) [inline includes/definitions/discovery/gaia.yaml:73]: and single quotes here.

**Review by murrant — CHANGES_REQUESTED** [review]

**Review by Jellyfrog — APPROVED** [review]

**Comment by chrisrnz** [comment]
This feature caused me some grief today, but I have a fleet of Check Point devices and a few different types in the mix.

The Clustered machines (pairs of 5600's) seemed to be happy after the sensors were added.

But then there were the dozens of branch office appliances (1430's) that all had the %macros.state_sensor_critical rule throwing alerts for "sensor_id = 544; sensor_oid = .1.3.6.1.4.1.2620.1.5.5.0;   sensor_descr = Cluster mechanism started; state_descr = no;"

Note: These are little embedded boxes (small office) style that run a different OS to the big ones.

The ha state sensors in health settings look like this to me.
Normal GAiA FW, clustered.
One member
state	haStarted	Cluster mechanism started	1	
state	haStatCode	Cluster member status code	0	
state	haState	Cluster member state	2
Other member
state	haStarted	Cluster mechanism started	1	
state	haStatCode	Cluster member status code	0	
state	haState	Cluster member state	1
Looks good

The little GAiA Embedded box, these are standalone, not clustered.
state	haStarted	Cluster mechanism started	0	
state	haStatCode	Cluster member status code	2

I'm happy to provide any further info that might be useful to help try fix this, just let me know what/how. :)

Cheers.
C.

**Pull request merged**